File: app/src/main/java/com/cybersiren/android/v2v/ui/V2VColors.kt
package com.cybersiren.android.v2v.ui

import androidx.compose.ui.graphics.Color

object V2VColors {

    val EmergencyRed = Color(0xFFFF3B30)
    val EmergencyRedDark = Color(0xFFCC2F26)
    val WarningOrange = Color(0xFFFF9500)
    val WarningOrangeDark = Color(0xFFCC7700)
    val SafeGreen = Color(0xFF34C759)
    val SafeGreenDark = Color(0xFF2A9F47)

    val AmbulanceBlue = Color(0xFF007AFF)
    val FireRed = Color(0xFFFF2D55)
    val PoliceBlue = Color(0xFF5856D6)
    val HazardYellow = Color(0xFFFFCC00)

    val UrgencyCritical = Color(0xFFFF3B30)
    val UrgencyHigh = Color(0xFFFF9500)
    val UrgencyMedium = Color(0xFFFFCC00)
    val UrgencyLow = Color(0xFF007AFF)

    val DarkBackground = Color(0xFF1C1C1E)
    val CardBackground = Color(0xFF2C2C2E)
    val CardBackgroundLight = Color(0xFF3C3C3E)

    val TextPrimary = Color(0xFFFFFFFF)
    val TextSecondary = Color(0xFFAEAEB2)
    val TextMuted = Color(0xFF8E8E93)

    val SenderModeColor = EmergencyRed
    val ReceiverModeColor = SafeGreen

    val StatusActive = SafeGreen
    val StatusInactive = TextMuted
    val StatusConnected = AmbulanceBlue

    val Accent = Color(0xFFFF5A1F)
    val AccentDark = Color(0xFFE64A10)
    val AccentSoft = Color(0xFFFFF1EA)

    val Ink = Color(0xFF0F1113)
    val InkSoft = Color(0xFF3A3F46)
    val Muted = Color(0xFF8A8F98)

    val BackgroundLight = Color(0xFFFBFBFC)
    val SurfaceLight = Color(0xFFFFFFFF)
    val BorderLight = Color(0xFFE6E8EC)

    val OnAccent = Color(0xFFFFFFFF)

    val Safe = Color(0xFF10B981)
    val SafeSoft = Color(0xFFECFDF5)

    val VehicleAmbulance = Color(0xFF2563EB)
    val VehicleFire = Color(0xFFDC2626)
    val VehiclePolice = Color(0xFF1E3A8A)
    val VehicleEmergency = Color(0xFFF59E0B)

    fun accentFor(type: com.cybersiren.android.v2v.model.VehicleType): Color = when (type) {
        com.cybersiren.android.v2v.model.VehicleType.AMBULANCE -> VehicleAmbulance
        com.cybersiren.android.v2v.model.VehicleType.FIRE_TRUCK -> VehicleFire
        com.cybersiren.android.v2v.model.VehicleType.POLICE_CAR -> VehiclePolice
        com.cybersiren.android.v2v.model.VehicleType.EMERGENCY -> VehicleEmergency
    }
}

File: ios/bitchat/bitchat/V2V/UI/V2VColors.swift
import SwiftUI

public enum V2VColors {

    public static let emergencyRed = Color(red: 1.00, green: 0.23, blue: 0.19)
    public static let warningOrange = Color(red: 1.00, green: 0.58, blue: 0.00)
    public static let safeGreen = Color(red: 0.20, green: 0.78, blue: 0.35)

    public static let urgencyCritical = emergencyRed
    public static let urgencyHigh = warningOrange
    public static let urgencyMedium = Color(red: 1.00, green: 0.80, blue: 0.00)
    public static let urgencyLow = Color(red: 0.00, green: 0.48, blue: 1.00)

    public static let accent = Color(red: 1.00, green: 0.35, blue: 0.12)
    public static let accentDark = Color(red: 0.90, green: 0.29, blue: 0.06)
    public static let accentSoft = Color(red: 1.00, green: 0.95, blue: 0.92)

    public static let ink = Color(red: 0.06, green: 0.07, blue: 0.07)
    public static let inkSoft = Color(red: 0.23, green: 0.25, blue: 0.28)
    public static let muted = Color(red: 0.54, green: 0.56, blue: 0.60)

    public static let backgroundLight = Color(red: 0.98, green: 0.98, blue: 0.99)
    public static let surfaceLight = Color.white
    public static let borderLight = Color(red: 0.90, green: 0.91, blue: 0.93)
    public static let onAccent = Color.white

    public static let safe = Color(red: 0.06, green: 0.72, blue: 0.51)
    public static let safeSoft = Color(red: 0.93, green: 0.99, blue: 0.96)

    public static let vehicleAmbulance = Color(red: 0.15, green: 0.39, blue: 0.92)
    public static let vehicleFire = Color(red: 0.86, green: 0.15, blue: 0.15)
    public static let vehiclePolice = Color(red: 0.12, green: 0.23, blue: 0.54)
    public static let vehicleEmergency = Color(red: 0.96, green: 0.62, blue: 0.04)

    public static func accentFor(_ type: VehicleType) -> Color {
        switch type {
        case .ambulance: return vehicleAmbulance
        case .fireTruck: return vehicleFire
        case .policeCar: return vehiclePolice
        case .emergency: return vehicleEmergency
        }
    }

    public static func urgencyColor(_ level: UrgencyLevel) -> Color {
        switch level {
        case .critical: return urgencyCritical
        case .high:     return urgencyHigh
        case .medium:   return urgencyMedium
        case .low:      return urgencyLow
        }
    }
}
